engine/src/main/scala/cromwell/engine/workflow/WorkflowStoreActor.scala

```diff
@@ -76,37 +76,35 @@ abstract class WorkflowStore {
   }
 
   def remove(id: WorkflowId): Boolean = {
-    val newWorkflowStore = workflowStore filterNot { _.id == id }
-
-    if (newWorkflowStore == workflowStore) {
-      false
-    } else {
-      workflowStore = newWorkflowStore
+    if (workflowStore.exists(_.id == id)) {
```

Comment:
In the worst case (the element exists but is the last one in the list) this is making two full scans of the list. If it doesn't exist it's the same amount of work (one full scan). In the best case (the match is the first element) it's making a scan + 1. 

Aside from my DB-izing comment this is small potatoes compared to how often we multi-scan collections all over the place which is another reason I don't really care, although that was my thinking when I did it that way in the first place, since the difference was the same LOC anyways